doc/protocol.md

```diff
@@ -282,15 +282,17 @@ Command: ping
 
 The "ping" command is simply a keep alive.
 
-No additional fields are defined.
+No additional fields are defined, but any are allowed. If a "channel"
+is set this "ping" will be forwarded. Otherwise it be limited to a single hop.
 
 An example of a ping:
 
     {
         "command": "ping",
     }
 
-Any protocol participant can send this message, but it is not responded to.
+Any protocol participant can send a "ping". It is responded to by sending
+a "pong" with identical options as a reply.
```

Comment:
This would technically a small API break, although I doubt that many/any external projects actually use ping messages, and moreover rely on not getting any reply. But according to the test case below the "pong" is only sent when giving a (valid) channel, and previously `ping` was only defined to be channel-less, so this is *not* an API break, right? The documentation should point that out (pong only sent with channel).

Reply:
The protocol states that "Unknown control messages are ignored". "pong" messages were previously an unknown message. Therefore this is not an API break.

"pong" can be sent both with and without a "channel" field. According to the documentation, control messages in channels are forwarded across the entire routing network as necessary, control messages without a channel are not forwarded but are a single hop.

"ping and "pong" can be sent either with or without a channel, to different effects.

Reply:
Both the main protocol docs in the "Control Message" section and the "Command: ping" section talk about this point. But I've tweaked the wording to be more clear.